integration_tests/test_gravity.py

```diff
@@ -440,52 +451,32 @@ def test_gravity_cancel_transfer(gravity):
         assert balance == 100000000000000000000000000
         amount = 1000
 
+        current_block = int(cli.status()["SyncInfo"]["latest_block_height"])
+        batch_block = 10
+        diff_block = batch_block - current_block % batch_block
+        wait_for_new_blocks(cli, diff_block)
+
         print("send to cronos crc21")
-        recipient = HexBytes(ADDRS["community"])
-        send_to_cosmos(gravity.contract, erc20, recipient, amount, KEYS["validator"])
+        community = HexBytes(ADDRS["community"])
+        send_to_cosmos(gravity.contract, erc20, community, amount, KEYS["validator"])
         assert erc20.caller.balanceOf(ADDRS["validator"]) == balance - amount
 
         denom = f"gravity{erc20.address}"
-
         crc21_contract = None
 
-        def check_auto_deployment():
-            "check crc21 contract auto deployed, and the crc21 balance"
+        def local_check_auto_deployment():
             nonlocal crc21_contract
-            try:
-                rsp = cli.query_contract_by_denom(denom)
-            except AssertionError:
-                # not deployed yet
-                return False
-            assert len(rsp["auto_contract"]) > 0
-            crc21_contract = cronos_w3.eth.contract(
-                address=rsp["auto_contract"], abi=cronos_crc21_abi()
+            crc21_contract = check_auto_deployment(
+                cli, denom, cronos_w3, community, amount
             )
-            return crc21_contract.caller.balanceOf(recipient) == amount
-
-        def get_id_from_receipt(receipt):
-            "check the id after sendToChain call"
-            for _, log in enumerate(receipt.logs):
-                if log.topics[0] == HexBytes(
-                    abi.event_signature_to_log_topic(
-                        "__CronosSendToChainResponse(uint256)"
-                    )
-                ):
-                    return log.data
-            return "0x0000000000000000000000000000000000000000000000000000000000000000"
-
-        wait_for_fn("send-to-crc20", check_auto_deployment)
-
-        def check_fund():
-            v = crc21_contract.caller.balanceOf(ADDRS["community"])
-            return v == amount
+            return crc21_contract
 
-        wait_for_fn("send-to-ethereum", check_fund)
+        wait_for_fn("send-to-crc21", local_check_auto_deployment)
 
         # send it back to erc20
         tx = crc21_contract.functions.send_to_chain(
```

Comment:
Better move the wait function here.  The transaction that we want to avoid being included in a batch is "send_to_chain"

Reply:
yeah, we should do following procedure for maximum stability:
- send-to-cosmos, and wait for balance
- wait for the edge block
- send-to-chain
- cancel

Reply:
it's already the case